tests/rados/test_mon_system_operations.py

```diff
@@ -0,0 +1,268 @@
+        )
+        raise Exception("Mon not in quorum error post addition")
+
+    # Writing Objects on pool post mon removal and OSD down
+    rados_obj.bench_write(pool_name=pool_name, byte_size=1024, rados_write_duration=50)
+
+    if not rados_obj.change_osd_state(action="start", target=osd_id):
+        log.error(f"Could not start OSD : {osd_id}")
+        raise Exception("OSD not started error")
+
+    # Now that the OSD is restarted, waiting for the recovery while the mon is removed
+    log.debug(
+        " waiting for PGs to settle down and recover post OSD stop and start during mon removal"
+    )
+    method_should_succeed(wait_for_clean_pg_sets, rados_obj, 5000)
+
+    if not rados_obj.run_pool_sanity_check():
+        log.error("Checks failed post mon removal and OSD rebalance")
+        raise Exception("Secnario 3 of Mon tests failed")
+
+    log.info("Completed scenario 3. Pass")
+
+    log.info(
+        "---------- starting workflow 4 - Addition of mons during OSD down --------"
+    )
+
+    # fetching all the OSDs on the cluster and selecting 1 OSD at random to be shut down.
+    out, _ = cephadm.shell(args=["ceph osd ls"])
+    osd_list = out.strip().split("\n")
+
+    osd_id = int(random.choice(osd_list))
+    if not rados_obj.change_osd_state(action="stop", target=osd_id):
+        log.error(f"Could not stop OSD : {osd_id}")
+        raise Exception("OSD not stopped error")
+
+    # Adding the mon back to the cluster
+    if not mon_obj.add_mon_service(host=test_mon_host):
+        log.error(f"Could not add mon service on host {test_mon_host}")
+        raise Exception("mon service not added error")
+
+    # Setting the mon service as managed by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=False):
+        log.error("Could not set the mon service to managed")
+        raise Exception("mon service not managed error")
+
+    # Checking if the mon added is part of Mon Quorum
+    quorum = mon_obj.get_mon_quorum_hosts()
+    if test_mon_host.hostname not in quorum:
+        log.error(
+            f"selected host : {test_mon_host.hostname} does not have mon as part of Quorum"
+        )
+        raise Exception("Mon not in quorum error")
+
+    # Writing Objects on pool post mon Addition and OSD down
+    rados_obj.bench_write(pool_name=pool_name, byte_size=1024, rados_write_duration=50)
+
+    if not rados_obj.change_osd_state(action="start", target=osd_id):
+        log.error(f"Could not start OSD : {osd_id}")
+        raise Exception("OSD not started error")
+
+    # Now that the OSD is restarted, waiting for the recovery while the mon is Added
+    log.debug(
+        " waiting for PGs to settle down and recover post OSD stop and start during mon Addition"
+    )
+    method_should_succeed(wait_for_clean_pg_sets, rados_obj, 5000)
+
+    if not rados_obj.run_pool_sanity_check():
+        log.error("Checks failed post mon removal and OSD rebalance")
```

Comment:
Error message should be for mon addition case.

Reply:
Updated the error message.